src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -111,20 +111,45 @@ public void setup() throws Exception {
 
 	@Test
 	public void testNameFilter() throws Exception {
-		// contract: legacy NameFilter is tested and works
+		// contract: the constructor of the class NameFilter creates and initialises an object.
+
 		CtClass<?> foo = factory.Package().get("spoon.test.filters.testclasses").getType("Foo");
 		assertEquals("Foo", foo.getSimpleName());
 		List<CtNamedElement> elements = foo.getElements(new NameFilter<>("i"));
-		// contract: tests NameFilter constructor for cases other than null
 		assertEquals(1, elements.size());
-		// contract: tests NameFilter constructor for null value case
-		assertThrows(IllegalArgumentException.class, () -> foo.getElements(new NameFilter<>(null)));
-		// contract: tests the getType method of the NameFilter class
-		assertEquals("CtNamedElement",new NameFilter<>("i").getType().getSimpleName());
-		// contract: tests the matches method of the NameFilter class
-		assertTrue(new NameFilter<>("i").matches(elements.get(0)));
 	}
 
+	@Test
+	void testNameFilterMatchesReturnsTrueForMatchingName() {
+		// contract: NameFilter.matches should return true for an element with a matching name
+
+		// arrange
+		String name = "someNiceName";
+		NameFilter<CtNamedElement> nameFilter = new NameFilter<>(name);
+		CtNamedElement elemWithMatchingName = new Launcher().getFactory().createLocalVariable();
+		elemWithMatchingName.setSimpleName(name);
+
+		// act
+		boolean matches = nameFilter.matches(elemWithMatchingName);
+
+		// assert
+		assertTrue(matches);
+	}
+
+	@Test()
+	public void testNameFilterThrowsExceptionForNull() {
+		// contract: NameFilter constructor should throw IllegalArgumentException when passed null
+
+		assertThrows(IllegalArgumentException.class, () -> new NameFilter<>(null));
+	}
+
+	@Test()
+	public void testNameFilterGetType() {
+		// contract: getType method returns the object of class NameFilter
```

Comment:
This contract is not correct, the `NameFilter.getType()` is not tested to return a `NameFilter` object.

Reply:
should the contract be like this : getType method returns the type of class which is begin used ?

Reply:
There's no one answer to what a contract should say, but it cannot be incorrect. What the `getType` method actually is supossed to return is the `CtNamedElement` _class_, which is not an object of class `NameFilter`, and thus the comment here is explicitly incorrect.

The contract comment should state in human-readable terms what the test is supposed to verify. One potential contract here would be: `// contract: NameFilter.getType() should return the CtNamedElement class`

A less specific one would be: `// contract: NameFilter.getType() should return the correct class`, where the test itself then specifies that correct class.

Note that we don't usually test getters directly, because they are trivial pieces of code, and so the test and its contract here necessarily become a bit contrived.

Reply:
> Note that we don't usually test getters directly, because they are trivial pieces of code

I have one question, so I can skip adding tests for getter like ```getFactory()``` in this [class](https://github.com/INRIA/spoon/blob/2fff4c50266f79bc06b961e3b01ca6c6e0fcc1ab/src/main/java/spoon/template/AbstractTemplate.java) ?

Reply:
> I have one question, so I can skip adding tests for getter like getFactory() in this class ?

Directly testing that would be fairly far down the priority list, yes. It would for example be much more important to test `spoon.template.Substitution.getFactory`, as that method is more complex. Overall, the `Substitution` class looks fairly poorly tested when looking at the [Coveralls report](https://coveralls.io/builds/40529606/source?filename=src%2Fmain%2Fjava%2Fspoon%2Ftemplate%2FSubstitution.java), so perhaps that'd be a good target for you!